tests/callbacks/test_rich_model_summary.py

```diff
@@ -11,25 +11,54 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from typing import Any
+from unittest import mock
+
 import pytest
+import torch
 
 from pytorch_lightning import Trainer
 from pytorch_lightning.callbacks import RichModelSummary, RichProgressBar
 from pytorch_lightning.utilities.imports import _RICH_AVAILABLE
+from pytorch_lightning.utilities.model_summary import summarize
+from tests.helpers import BoringModel
 from tests.helpers.runif import RunIf
 
 
 @RunIf(rich=True)
 def test_rich_model_summary_callback():
-
     trainer = Trainer(callbacks=RichProgressBar())
 
     assert any(isinstance(cb, RichModelSummary) for cb in trainer.callbacks)
     assert isinstance(trainer.progress_bar_callback, RichProgressBar)
 
 
 def test_rich_progress_bar_import_error():
-
     if not _RICH_AVAILABLE:
         with pytest.raises(ImportError, match="`RichModelSummary` requires `rich` to be installed."):
             Trainer(callbacks=RichModelSummary())
+
+
+@RunIf(rich=True)
+@mock.patch("pytorch_lightning.callbacks.rich_model_summary.Console.print", autospec=True)
+@mock.patch("pytorch_lightning.callbacks.rich_model_summary.Table.add_row", autospec=True)
+def test_rich_summary_tuples(mock_table_add_row, mock_console):
```

Comment:
@SeanNaren shouldn't a test be added to the model summary utility tests?

Perhaps there was a confusion here, there are different ModelSummaries, one is a callback the other is a normal class from utilities:

```
pytorch_lightning.utilities.model_summary.ModelSummary
pytorch_lightning.callbacks.model_summary.ModelSummary
```

I would argue this change should be tested universally for the utility ModelSummary

Reply:
thanks @awaelchli yes it should be, was curious if this should be tested through the callback ModelSummary or through the utility as it's a protected function. I much prefer just testing the protected function as this is much simpler